File: src/main/java/com/example/rms/business/category/CategoryService.java
package com.example.rms.business.category;

import com.example.rms.business.category.dto.CategoryDTO;
import com.example.rms.business.category.dto.CategoryWithThreads;
import com.example.rms.business.category.dto.CategoryRequest;

import java.util.List;

public interface CategoryService {
    List<CategoryDTO> getCategories();
    CategoryWithThreads getCategoryDetail(Integer categoryId);
    CategoryDTO createCategory(CategoryRequest request);
    CategoryDTO updateCategory(CategoryRequest request, Integer categoryId);
    void deleteCategory(Integer categoryId);
}

File: ui/src/services/CategoryService.js
import api from "../utils/api"

const basePath = "/category"
export default {
    getCategories: () => api.get(`${basePath}`),
    getCategory: (categoryId) => api.get(`${basePath}/${categoryId}`),
    createCategory: (request) => api.post(`${basePath}`, request),
    updateCategory: (categoryId, request) => api.put(`${basePath}/${categoryId}`, request),
    deleteCateogry: (categoryId) => api.delete(`${basePath}/${categoryId}`)
}